### PR TITLE
Add config to set maintenance_window

### PR DESCRIPTION
**Pull request by mfarix**
Based on the following documents:

- [Terraform Registry - Cloud SQL Module](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance)
- [Github - Cloud SQL Module](https://github.com/terraform-google-modules/terraform-google-sql-db/blob/master/modules/mssql/README.md)

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -267,3 +267,21 @@ variable "additional_databases" {
...
+  EOT
+  type = object({
+    day_utc      = number
+    hour_utc     = number
+    update_track = string
+  })
+  default = {
+    day_utc      = 1  # implies Tuesday in MYT
+    hour_utc     = 19 # implies 3AM in MYT
+    update_track = "canary"
```
Comment (syedrakib) [inline variables.tf]: By default, we should recommend `stable`

File: variables.tf [hunk 2]
```diff
@@ -267,3 +267,21 @@ variable "additional_databases" {
   }))
   default = []
 }
+
+variable "maintenance_window" {
+  description = <<-EOT
+  day: The day of week (1-7), starting on Monday.
```
Comment (syedrakib) [inline variables.tf]:
```suggestion
  day_utc: The day of the week (1-7) in UTC timezone - starting from Monday.
```

File: variables.tf [hunk 3]
```diff
@@ -267,3 +267,21 @@ variable "additional_databases" {
...
+  hour: The hour of day (0-23), ignored if day not set.
```
Comment (syedrakib) [inline variables.tf]:
```suggestion
  hour_utc: The hour of the day (0-23) in UTC timezone - ignored if day is not set.
```

File: variables.tf [hunk 4]
```diff
@@ -267,3 +267,21 @@ variable "additional_databases" {
...
+  update_track: The update track of maintenance window. Can be either `canary` or `stable`.
```
Comment (syedrakib) [inline variables.tf]:
```suggestion
  update_track: The update track of maintenance window - can be either `canary` or `stable`.
```

File: variables.tf [hunk 5]
```diff
@@ -267,3 +267,21 @@ variable "additional_databases" {
...
+  EOT
+  type = object({
+    day_utc      = number
+    hour_utc     = number
+    update_track = string
+  })
+  default = {
+    day_utc      = 1  # implies Tuesday in MYT
```
Comment (syedrakib) [inline variables.tf]: Actually, this comment (when read by itself) is misleading. `day_utc = 1` does not imply Tuesday in MYT. Rather, it's more like `day_utc = 1 AND time_utc > 18` implies Tuesday MYT.

To illustrate what you tried to say here, you can write this full matter in the description itself at its end. So users of the module will know from the doc what is expected.

File: variables.tf [hunk 6]
```diff
@@ -267,3 +267,21 @@ variable "additional_databases" {
...
+    hour_utc     = 19 # implies 3AM in MYT
```
Comment (syedrakib) [inline variables.tf]: Can remove these comments if you are updating the description with the UTC <> MYT matter i mentioned above

**Review by syedrakib — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (mfarix) [reply variables.tf]: This config is useful when there is more than one SQL instance in the same project. Lets say one for staging and another for production.
The config indicates to receive updates earlier (canary) or later (stable). So you would ideally set `canary` for staging and `stable` for production. So staging DB will get updated first before production DB update.

According to [GCP docs](https://cloud.google.com/sql/docs/mysql/instance-settings?hl=en#maintenance-timing-2ndgen):
> The relative timing of updates is not observed between projects; if you have instances with an earlier timing setting in a different project than your instances with a later timing setting, Cloud SQL makes no attempt to update the instances with the earlier timing setting first.

**Comment by syedrakib** [comment]
LGTM

**Review by sundarsudandiraraj — APPROVED** [review]
Useful feature. Thanks

**Review by syedrakib — APPROVED** [review]

**Pull request merged**